hack/tools/tools.go

```diff
@@ -0,0 +1,25 @@
+//go:build tools
+// +build tools
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This package imports things required by build scripts, to force `go mod` to see them as dependencies
+package tools
+
+import (
+	_ "sigs.k8s.io/cluster-api/hack/tools"
```

Comment:
do we need this? why not install it with go install how we install others like envsubst?

Reply:
The CAPI hack/tools package has a `replace` statement in it with a fake SHA that prevents `go install` (or `go get`) from working. Adding a local package in CAPZ and a local go.mod seems like the only solution, at least that's where I got to as well when looking at this.